docs/components.md

```diff
@@ -265,48 +174,22 @@ function ComponentWithState() {
 }
 ```
 
-A big advantage of closure components is that we don't need to worry about binding `this` when attaching event handler callbacks. In fact `this` is never used at all and we never have to think about `this` context ambiguities.
+Closure components are consumed in the same way as POJOs, e.g. `m(ComponentWithState, { passedData: ... })`.
 
-#### Class Component State
+A big advantage of closure components is that we don't need to worry about binding `this` when attaching event handler callbacks. In fact `this` is never used at all and we never have to think about `this` context ambiguities.
 
-With classes, state can be managed by class instance properties and methods. For example:
 
-```javascript
-class ComponentWithState() {
-	constructor() {
-		this.count = 0
-	}
-	increment() {
-		this.count += 1
-	}
-	decrement() {
-		this.count -= 1
-	}
-	view() {
-		return m("div",
-			m("p", "Count: " + count),
-			m("button", {
-				onclick: () => {this.increment()}
-			}, "Increment"),
-			m("button", {
-				onclick: () => {this.decrement()}
-			}, "Decrement")
-		)
-	}
-}
-```
-
-Note that we must wrap the event callbacks in arrow functions so that the `this` context is preserved correctly.
+---
 
 #### POJO Component State
 
-For POJO components the state of a component can be accessed three ways: as a blueprint at initialization, via `vnode.state` and via the `this` keyword in component methods.
+It is generally accepted that a component with state that must be managed is best expressed as a closure. If, however, you have reason to manage state in a POJO, the state of a component can be accessed in three ways: as a blueprint at initialization, via `vnode.state` and via the `this` keyword in component methods.
```

Comment:
I'd like @pygy and maybe @StephanHoyer to chime in on this one. I'd prefer to not introduce *this* strong of an opinion against a particular *supported* component type when our opinion is already pretty clearly implied through the broad use of closure components instead. Also, [citation needed] for the "It is generally accepted that ..." part - I'm not aware of anyone who's done a survey to verify that.

Reply:
Something shorter and less contentious:

> Whereas state management in closure components is a simple matter of scoped reference, in the case of a component defined as an object,

?

Reply:
I'm great with others chiming in, but I strongly disagree that this is a "strong...opinion against" the use of component state in POJOs. This is a strong advocacy _for_ using closures ("best expressed as..."), but there's nothing negative suggested to the contrary, nor are there any of the warning flags we've been joking about on gitter.

I believe this is the kind of confident opinion which our docs are sorely lacking, TBF. This is proper _guidance_, essentially in the same form which we all use ad nauseam on gitter when someone asks. 

Scenario: A random person pops in and says, "I have a component which needs to handle some internal state. What's the best approach?" 

My hypothesis: About ten people — @isiahmeadows and @barneycarroll included — will jump in and recommend a closure (within minutes, if not moments). 

Of course there's been no survey. But the easily-verified evidence on our gitter channel, from those in the community who are most engaged with the framework, speaks volumes.

I think the docs should have the courage of our convictions, rather than strive to achieve some sort of consensus. It's a form of design/write/code by committee which is too broad and never ends well.

Reply:
Personally, I'd just like to see this mostly reverted with the grammatical issues fixed:

```suggestion
For POJO components, the state of a component can be accessed in three ways: as a blueprint at initialization, via `vnode.state` and via the `this` keyword in component methods.
```

The language is inconsistent with the rest of the docs in how opinionated it is. A reader who's not very familiar with Mithril would be wondering why the feature is even there if we're heavily discouraging its use like that. Consider that those kinds of people are going to be reading this just as much as someone who's already actively using it.

I could totally go with it [in a blog post](https://github.com/MithrilJS/mithril.js/issues/2101), though. And I do agree [we could do better](https://github.com/MithrilJS/mithril.js/issues/2295) in a breaking change.

Reply:
Better yet, you could recommend that POJO components should be used for *stateless* components. Something like this:

> It's generally recommended that POJO components should be stateless. If, however, you need to access and/or modify a POJO component's state, it can be accessed in three ways: as a blueprint at initialization, via `vnode.state`, and via the `this` keyword in component methods.

This way, it's a little more positive and it doesn't discourage the common case of embedding read-only data within the component itself, but it more clearly recommends *when* it's most useful.

Reply:
I also wouldn't mind classes being recommended for TS+JSX users (thanks to https://github.com/Microsoft/TypeScript/issues/21699 and https://github.com/Microsoft/TypeScript/issues/14789).

Reply:
I'd probably agree, but @StephanHoyer will not, since lifecycle hooks are stateful.

Reply:
...as for TS + JSX, I'm not qualified to talk about it.

Reply:
@CreaturesInUnitards Maybe stateful relative to the DOM, but I'm talking stateful relative to the component itself. If your DOM integration only requires something like `$(vnode.dom).plugin({...})` and you can just update it with, say, `$(vnode.dom).plugin("action")` (this is the case with Bootstrap, BTW), then you don't need to care about `vnode.state`.

Also, not all hooks have to be stateful:

- `onbeforeupdate` serves the secondary purpose of allowing you to diff attributes and cancel updates.
- `onbeforeremove` serves to let you do things like CSS transitions.
- `view` is obvious.

Reply:
I'm fine with @CreaturesInUnitards proposal. As we now move away from POJO and class components (which I strongly support, although I currently only use POJOs for historic reasons), it might be ok to make POJO-Components less prominent.

As stated on Gitter, I'm against the notion of `stateless components` and the relation to POJO Components.

Reply:
Okay. I'm personally a little uncomfortable with the wording, but I'll go with it if everyone else is okay with it.